specification/signalr/data-plane/AzureSignalR/v1/azuresignalr.json

```diff
@@ -0,0 +1,875 @@
+            "type": "integer",
+            "format": "int32"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success"
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "type": "string"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "signalr.v1"
+        ],
+        "summary": "Remove a user from the target group.",
+        "operationId": "HubProxyControllerV1RemoveUserFromGroup_Delete",
+        "produces": [
+          "text/plain"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "hub",
+            "description": "Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "group",
+            "description": "Target group name, which length should be greater than 0 and less than 1025.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "user",
+            "description": "Target user Id",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success"
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "/api/v1/hubs/{hub}/users/{user}/groups": {
+      "delete": {
+        "tags": [
+          "signalr.v1"
+        ],
+        "summary": "Remove a user from all groups.",
+        "operationId": "HubProxyControllerV1RemoveUserFromAllGroups_Delete",
+        "produces": [
+          "text/plain"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "hub",
+            "description": "Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "user",
+            "description": "Target user Id",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The user is deleted"
+          },
+          "202": {
+            "description": "The delete request is accepted and service is handling the request int the background"
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PayloadMessage": {
```

Comment:
Is anything required, or all is optional?

Reply:
All the payload messages are required

Reply:
I meant, are `target` and `arguments` required for a `PayloadMessage` structure?

Reply:
nice catch! `target` should be `required`